### PR TITLE
Design change

### PR DESCRIPTION
**Pull request by adityacp**

### CODE REVIEW

File: yaksh/views.py [hunk 1]
```diff
@@ -859,7 +859,9 @@ def get_questions(paper):
         q_unanswered = paper.get_unanswered_questions()
         q_unanswered.sort()
         to_attempt = q_unanswered
-    for index, value in enumerate(all_questions, 1):
+    
+    question_objs = Question.objects.filter(id__in=all_questions)
```
Comment (prathamesh920) [inline yaksh/views.py]: You can remove _objs from the above variable name.
Can give other name, also note that you already have questions variable below.

File: yaksh/views.py [hunk 2]
```diff
@@ -1024,7 +1027,7 @@ def check(request, q_id, attempt_num=None, questionpaper_id=None):
                    'paper': paper, 'last_attempt': user_code,
                    'quiz_name': paper.question_paper.quiz.description,
                    'time_left': time_left, 'questions': questions,
-                   'to_attempt': to_attempt, 'submitted': submitted}
+                   'to_attempt': to_attempt, 'submitted': submitted, 'answer':Answer}
```
Comment (prathamesh920) [inline yaksh/views.py]: There should be space after the colon.
Use PEP8 for this purpose.

File: yaksh/templates/yaksh/question.html [hunk 3]
```diff
@@ -183,10 +184,9 @@ <h3 class="brand"><strong>Online Test</h3></strong>
...
-
-         <textarea rows="1" style="padding:0;height:auto;width:750px;overflow:hidden;background-color:white;border: 0 none white;" readonly="yes"  name="snippet" id="snippet" wrap="off">{% if last_attempt %}{{ question.snippet }}{% else %}{% if question.type == "bash" %} #!/bin/bash&#13;&#10;{{ question.snippet }}{% else %}{{ question.snippet }}{% endif %}{% endif %}</textarea>
-        
-         <textarea tabindex=1 rows="10" style="padding:0;height:auto; box-shadow: none;width:750px;margin-bottom:10px;overflow:hidden;border:none;" name="answer" id="answer" wrap="off" onkeydown="return catchTab(this,event)">{% if last_attempt %}{{last_attempt}}{% else %}{% if question.type == "bash" %}{% else %}{% endif %}{% endif %}</textarea>
+        <h3>Program:</h3>
+        <textarea rows="1" style="padding:0;height:auto;width:750px;resize:None;overflow:hidden;background-color:white;border: 0 none white;" readonly="yes"  name="snippet" id="snippet" wrap="off" >{% if last_attempt %}{{ question.snippet }}{% else %}{% if question.type == "bash" %} #!/bin/bash&#13;&#10;{{ question.snippet }}{% else %}{{ question.snippet }}{% endif %}{% endif %}</textarea>
```
Comment (prabhuramachandran) [inline yaksh/templates/yaksh/question.html]: While I know this is not your code, this bash line here is very dirty and ugly, please refactor this to remove it from the template and do not hardcode these.

File: yaksh/views.py (hunk 2 above)
Comment (prabhuramachandran) [inline yaksh/views.py]: Also what is this answer doing here?  This is not clear.

File: yaksh/templates/yaksh/question.html [hunk 4]
```diff
@@ -55,6 +56,9 @@
     if (t.total>=300000) {
       clock.innerHTML = "<strong>" + hh + ":" + mm + ":" + ss + "</strong>";
     }
+    if (t.total==0){
+		alert("Your exam is submitted successfuly");
```
Comment (prabhuramachandran) [inline yaksh/templates/yaksh/question.html]: Alert is ugly.

Comment (prabhuramachandran) [inline yaksh/templates/yaksh/question.html]: Remove this.

File: yaksh/templates/yaksh/question.html [hunk 5]
```diff
@@ -93,33 +97,25 @@
...
         <h3 class="brand"><strong>Online Test</h3></strong>
         <ul>
-        <li> <h5><a> Hi {{user.first_name.title}} {{user.last_name.title}}  </a></h5>
+        <li> <h5><a> Hi {{user.first_name.title}} {{user.last_name.title}}  </a></h5></li>
         </ul>
         <form id="logout" action="{{URL_ROOT}}/exam/quit/{{ paper.attempt_number }}/{{ paper.question_paper.id }}/" method="post" class="pull-right">
         {% csrf_token %}
-         <button class="btn" type="submit" name="quit">Quit Exam</button> </li>
-
+        <button class="btn" type="submit" name="quit">Quit Exam</button>
```
Comment (prabhuramachandran) [inline yaksh/templates/yaksh/question.html]: Quit should be somewhere else, preferably at the bottom.

File: yaksh/templates/yaksh/question.html [hunk 6]
```diff
@@ -149,19 +145,24 @@ <h3 class="brand"><strong>Online Test</h3></strong>
...
+  <br><h3>Output:</h3></br>
+    {% if error_message %}
+    <div class="alert alert-error">
+      <textarea style="padding:40px;height:auto;width:650px;resize:None;overflow-y:scroll;background-color:white;border: 0 None white;" readonly="yes" >
+        {% for e in error_message.splitlines %}
+        {{ e|join:"" }} 
+        {% endfor %}
+		</textarea>
+	{% else %}
+	  <textarea style="padding:50px;height:auto;width:650px;resize:None;overflow:hidden;background-color:white;border: 12px solid #fcf8e3;" readonly="yes"  placeholder="Output will be shown here!!!"></textarea>
```
Comment (prabhuramachandran) [inline yaksh/templates/yaksh/question.html]: default placeholder should be "Please submit your answer."

**Pull request closed without merging**